eth/tracers/api.go

```diff
@@ -534,7 +541,14 @@ func (api *API) IntermediateRoots(ctx context.Context, hash common.Hash, config
 			vmenv     = vm.NewEVM(vmctx, txContext, statedb, chainConfig, vm.Config{})
 		)
 		statedb.Prepare(tx.Hash(), i)
-		if _, err := core.ApplyMessage(vmenv, msg, new(core.GasPool).AddGas(msg.Gas())); err != nil {
+
+		l1DataFee, _, _, err := fees.CalculateFees(tx, statedb)
+		if err != nil {
+			log.Warn("Tracing intermediate roots did not complete due to fees.CalculateFees", "txindex", i, "txhash", tx.Hash(), "err", err)
+			return roots, nil
```

Comment:
ditto

Reply:
fixed https://github.com/scroll-tech/go-ethereum/pull/351/commits/71b775d03efd523f32b9d543be379f4b66b1cbe2

Reply:
i was thinking about this, and thought I'd better "intentionally do not return err" as below.

but after rethinking about it i agree we should return error